src/SdfEntityCreator.cc

```diff
@@ -788,8 +793,32 @@ Entity SdfEntityCreator::CreateEntities(const sdf::Visual *_visual)
   // \todo(louise) Populate with default material if undefined
   if (_visual->Material())
   {
+    sdf::Material visualMaterial = *_visual->Material();
+    if (!_visual->Material()->ScriptUri().empty())
+    {
+      std::string scriptUri = visualMaterial.ScriptUri();
+      if (scriptUri != "file://media/materials/scripts/gazebo.material") {
+        gzwarn << "Custom material scripts are not supported."
+          << std::endl;
+      }
+    }
+    if (!_visual->Material()->ScriptName().empty())
+    {
+      std::string scriptName = visualMaterial.ScriptName();
+
+      if((scriptName.find("Gazebo/") != std::string::npos)) {
+        gzwarn << "Using an internal gazebo.material to parse " << scriptName
+          << std::endl;
+        this->dataPtr->materialParser.Load();
+        MaterialParser::MaterialValues parsed = this->dataPtr->materialParser.GetMaterialValues(scriptName);
+
+        visualMaterial.SetAmbient(parsed.ambient.value());
```

Comment:
I think we need to do the following for each color type:
```suggestion
        if (parsed.ambient.has_value())
        {
          visualMaterial.SetAmbient(parsed.ambient.value());
        }
```
Otherwise, it could throw an exception if that particular color type is missing from the material.

Reply:
Would value_or suit this case as we could give a default color?

Reply:
Sure. In that case, I would use the default color from the material itself. 
```c++
visualMaterial.SetAmbient(parsed.ambient.value_or(visualMaterial.Ambient()));
```
This makes sure that if we change the default colors at some point in sdformat, we wouldn't have to change it here.